### DISCUSSION TITLE
Multiple swagger docs, single config?

### DISCUSSION THREAD

**Discussion by katerinatiddy** [orval-labs/orval#722] [Q&A]
I have over 100 different API endpoints on my backend, each with a separate Swagger doc. Is there a way within the config to find all the swagger docs, perhaps with a wildcard, and generate types for them without having to manually type each one out?

**Comment by gavinthomas-valtech** [comment]
I was also wondering this. Im going to need multiple openapi.yaml files. Can i use multiuple orval config files?

**Reply by katerinatiddy** [reply]
I found a way around it by having a separate shell script I run that looks for all openapi.yaml files and calls orval in a loop with the input files as args. Then inside the orval.config.js I have something along the lines of:
```
const process = require('process');
const INPUT_PATH = process.argv[2]
...
module.exports = {
  'TEST API': {
    input: {
      target: `${INPUT_PATH}`,
    }, ...
}
```
Works quite well but does take some time with over 100 openapi.yaml files. I hope this helps though!

**Reply by gavinthomas-valtech** [reply]
thankyou. So target can take an array?

**Reply by anymaniax** [reply]
You can directly do that in the orval config which can take a function. So with a base config and an array with the specifications path and you just create the all config with a loop on the specifications array

**Reply by gavinthomas-valtech** [reply]
>You can directly do that in the orval config which can take a function

i might need to see a code example to get my head around this. from the spec it says input target takes a string.

>const INPUT_PATH = process.argv[2]

What is this part doing?

**Reply by anymaniax** [reply]
Next week I will add an example for that.

**Reply by gavinthomas-valtech** [reply]
amazing! thankyou very much. I've been assessing a lot of codegen tools for apis and this one is winning!

**Reply by gavinthomas-valtech** [reply]
heard about it from here https://profy.dev/article/react-openapi-typescript

**Reply by gavinthomas-valtech** [reply]
Any luck with that example @anymaniax ?

**Reply by anymaniax** [reply]
Sure on my current project I do it like [this](https://gist.github.com/anymaniax/70ee638783470ac90d224417e32c0412)

**Reply by anymaniax** [reply]
but you can also give an async function to the `defineConfig` function if you want and it another way

**Reply by gavinthomas-valtech** [reply]
this is great. Thankyou!

**Comment by katerinatiddy** [comment]
@gavinthomas-valtech To clarify what I said earlier here is a better example of what I currently have set up:

I have a script called generate-orval.sh which I call in the command line for example. Inside has the following:
```
#!/bin/bash
for INPUT_PATH in $(find /home/user/repo/api/ -name "*.yml")
do
    echo "Generating types for $(basename -s .yml ${INPUT_PATH})"
    orval "${INPUT_PATH}"
done
```
This loops around my repo's api folder looking for my yml files which are my openapi specs and calls orval with the path of that file as an argument. 
Then in the orval.config.js I have:
```
const process = require('process');
const INPUT_PATH = process.argv[2]

module.exports = {
  'TEST API': {
    input: {
      target: `${INPUT_PATH}`,
    }, ... 
  }
```
The first and second lines are taking the arguments into the orval config. This is then used in the target and runs against that specific yaml file I passed in. Some more info about process.argv can be found [here](https://stackoverflow.com/questions/22213980/could-someone-explain-what-process-argv-means-in-node-js-please).
